ultralytics/engine/predictor.py

```
@@ -162,7 +162,9 @@ def preprocess(self, im: Union[torch.Tensor, List[np.ndarray]]) -> torch.Tensor:
             im = np.stack(self.pre_transform(im))
             if im.shape[-1] == 3:
                 im = im[..., ::-1]  # BGR to RGB
-            im = im.transpose((0, 3, 1, 2))  # BHWC to BCHW, (n, 3, h, w)
+            elif im.shape[-1] == 1 or im.shape[0] == 1 or im.ndim == 2:
+                im = np.expand_dims(im, -1) # Grayscale (H, W) to (H, W, 1)
```

Flake8 (7.2.0)

[error] at least two spaces before inline comment

(E261)